p2p/discovery/option_test.go

```diff
@@ -0,0 +1,90 @@
+package discovery
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	p2ptypes "github.com/harmony-one/harmony/p2p/types"
+	libp2p_peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+var (
+	tmpDir    = filepath.Join(os.TempDir(), "harmony-one", "harmony", "p2p", "discovery")
+	emptyFile = filepath.Join(tmpDir, "empty_file")
+	validPath = filepath.Join(tmpDir, "dht-1.1.1.1")
+)
+
+var (
+	testAddrStr = []string{
+		"/ip4/52.40.84.2/tcp/9800/p2p/QmbPVwrqWsTYXq1RxGWcxx9SWaTUCfoo1wA6wmdbduWe29",
```

Comment:
can you test dns based boot nodes?

Reply:
Currently libp2p does not support DNS based boot node. The DNS address is already parsed in `main.go`